docs/portingguide/src/howto/nogc.md

```diff
@@ -1,50 +1,283 @@
+
+You should use the saved `Mutator` pointer as the first parameter, the requested object size as the next parameter, and any alignment requirements the runtimes has as the third parameter.
+
+If your runtime requires a non-zero allocation offset (i.e. returned address vs actual allocated address) then you have to provide the required value as the fourth parameter. Note that you ***must*** also update the [`USE_ALLOCATION_OFFSET`](https://www.mmtk.io/mmtk-core/public-doc/vm/trait.VMBinding.html#associatedconstant.USE_ALLOCATION_OFFSET) constant in the `VMBinding` implementation if your runtime requires a non-zero allocation offset.
+
+For the time-being, you can ignore the `allocator` parameter in both these functions and always pass a value of `0` which means MMTk will pick the default allocator for your collector (a bump pointer allocator in the case of NoGC).
+
+Finally, you need to call `mmtk_post_alloc` with the object address returned from the previous `mmtk_alloc` call in order to initialize object metadata.
+
+**Note:** Currently MMTk faultily assumes object sizes are multiples of the `VM::MIN_ALIGNMENT`. If you encounter errors with alignment, a simple workaround would be to align the requested object size up to the `VM::MIN_ALIGNMENT`. See [here](https://github.com/mmtk/mmtk-core/issues/730) for the tracking issue to fix this bug.
+
+### Rust changes
+
+The Rust side of the binding should simply defer the actual implementation to [`mmtk::memory_manager::alloc`](https://www.mmtk.io/mmtk-core/public-doc/memory_manager/fn.alloc.html) and [`mmtk::memory_manager::post_alloc`](https://www.mmtk.io/mmtk-core/public-doc/memory_manager/fn.post_alloc.html) respectively. See the [OpenJDK](https://github.com/mmtk/mmtk-openjdk/blob/54a249e877e1cbea147a71aafaafb8583f33843d/mmtk/src/api.rs#L125-L136) [binding](https://github.com/mmtk/mmtk-openjdk/blob/54a249e877e1cbea147a71aafaafb8583f33843d/mmtk/src/api.rs#L151-L161) for an example.
+
+Congratulations! At this point, you hopefully have object allocation working and can run simple programs with your runtime using MMTk!
+
+## Miscellaneous implementation steps
+
+### Setting options for MMTk
+
+You can set [options for MMTk](https://www.mmtk.io/mmtk-core/public-doc/util/options/index.html) by using the [`process`](https://www.mmtk.io/mmtk-core/public-doc/memory_manager/fn.process.html) function to pass options, or simply by setting environment variables. For example, to use the NoGC plan, you can set the environment variable `MMTK_PLAN=NoGC`. You may want to set multiple options at the same time as well. In such a case you can use the [`process_bulk`](https://www.mmtk.io/mmtk-core/public-doc/memory_manager/fn.process_bulk.html) function.
```

Comment:
A preferred way to set options is simply use `mmtk.options.x.set()`. You should mention that you can set options by env vars but that is not recommended. We recommend setting options through command line arguments. Env vars may be useful before command line arguments is properly implemented.

Reply:
That can only be set on the Rust side though? Command-line options are often done on the C/C++ side.

Reply:
Also aren't all options supposed to be set before MMTk initialization? That was my recollection, but maybe it's been fixed.

Reply:
> That can only be set on the Rust side though? Command-line options are often done on the C/C++ side.

Not all bindings are implemented in C/C++. They can be implemented in Rust.

> Also aren't all options supposed to be set before MMTk initialization? That was my recollection, but maybe it's been fixed.

Yeah. My bad. I meant: `builder.options.x.set()`, see https://github.com/mmtk/mmtk-openjdk/blob/54a249e877e1cbea147a71aafaafb8583f33843d/mmtk/src/api.rs#L79.

Reply:
Right I'll mention it then :+1: